patternfly/patternfly-elements
# feat: pfe-clipboard

## pfe-clipboard

pfe-clipboard allows developers to include button to "copy url to clipboard" that can be placed web pages and handle the entire process of displaying a button that, when clicked, will copy the url to the users system clipboard.

### Related Issues

- #803
- #810 (possibly)

### Preview

Link(s) to demo page(s) where this element can be viewed:
- [Link](https://deploy-preview-1235--happy-galileo-ea79c4.netlify.app/elements/pfe-clipboard/demo/) 

### Testing instructions



1. Open the pfe-clipboard demo page in your browser.
2. In your url bar, append a unique query parameter`?testing123`
3. Click on one of the 'Copy Url' buttons
4. Open a separate browser tab and paste your clipboard into the url
5. You should see the url of the original demo page as well as the unique query parameter
- Example: `https://deploy-preview-1235--happy-galileo-ea79c4.netlify.app/elements/pfe-clipboard/demo/?testing123`


#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist

Check off items as they are completed.  Feel free to delete items if they are not applicable.



### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
